core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -117,15 +117,20 @@ class BaseColumnFormatRelation(
   override def buildUnsafeScan(requiredColumns: Array[String],
   filters: Array[Filter]): (RDD[Any], Seq[RDD[InternalRow]]) = {
     val (rdd, _) = scanTable(table, requiredColumns, filters)
-
     val zipped = buildRowBufferRDD(rdd.partitions, requiredColumns, filters, true).
         zipPartitions(rdd) { (leftItr, rightItr) =>
           Iterator[Any](leftItr, rightItr)
         }
-
     (zipped, Nil)
   }
 
+  def buildUnsafeScanForSampledRelation(requiredColumns: Array[String],
+                               filters: Array[Filter]): (RDD[Any], Seq[RDD[InternalRow]]) = {
+    val (rdd, _) = scanTable(table, requiredColumns, filters)
+    (rdd.asInstanceOf[RDD[Any]], Nil)
```

Comment:
We have left rows in rowsbuffer (though a minuscule amount). Please create a ticket to handle this later. Can comment out code which still put some rows in rows buffer.